tests/unit_tests/test_ladder_service.py

```diff
@@ -139,28 +151,44 @@ async def test_start_game_timeout(
     assert p2.state is PlayerState.IDLE
 
 
+@pytest.mark.filterwarnings("ignore:.*uses the '.*' fixture")
```

Comment:
What is this about?

Reply:
Hypothesis raises a warning when you use function scoped fixtures because they will only be initialized once, and not every time hypothesis queries the test function. You have to make sure that your test function will handle cleaning up its own state changes if it makes any. In our case, it doesn't make any except for mocking certain function calls, which is perfectly fine to do over and over without resetting the fixtures.